crates/hyperqueue/src/server/autoalloc/state.rs

```diff
@@ -139,6 +152,18 @@ impl QueueState {
         }
     }
 
+    pub fn state(&self) -> &AllocationQueueState {
+        &self.state
+    }
+
+    pub fn pause(&mut self) {
+        self.state = AllocationQueueState::Paused;
+    }
+
+    pub fn resume(&mut self) {
+        self.state = AllocationQueueState::Running;
```

Comment:
I probably do not understand this well, but I would expected here some kind of "wake up" that tests if we should not (almost) immediately ask for new allocations.

Reply:
Good point. It's not strictly needed, but it's definitely useful. I added a refresh of the queue once it's resumed.